# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/security/nsCSPUtils.cpp
# Commit: 55f13fb4ee3f
# Full Hash: 55f13fb4ee3f06209d00fd91a7a03884bab88e2d
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-12-15 09:27:59
# Regressor Bug: 1247687
# File Overlap Count: 1
# Description:
#   Bug 1247687 - Implement csp for Module Workers; r=evilpie,asuth,rpl,ckerschb
#   
#   Depends on D155691
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D156102
# ==============================================================================

diff -r 354711cf113a -r 55f13fb4ee3f dom/security/nsCSPUtils.cpp
--- a/dom/security/nsCSPUtils.cpp	Wed Dec 14 14:55:37 2022 +0000
+++ b/dom/security/nsCSPUtils.cpp	Wed Dec 14 14:55:38 2022 +0000
@@ -304,6 +304,7 @@
       return nsIContentSecurityPolicy::WEB_MANIFEST_SRC_DIRECTIVE;
 
     case nsIContentPolicy::TYPE_INTERNAL_WORKER:
+    case nsIContentPolicy::TYPE_INTERNAL_WORKER_STATIC_MODULE:
     case nsIContentPolicy::TYPE_INTERNAL_SHARED_WORKER:
     case nsIContentPolicy::TYPE_INTERNAL_SERVICE_WORKER:
       return nsIContentSecurityPolicy::WORKER_SRC_DIRECTIVE;
@@ -353,6 +354,7 @@
     // Fall through to error for all other directives
     // Note that we should never end up here for navigate-to
     case nsIContentPolicy::TYPE_INVALID:
+    case nsIContentPolicy::TYPE_END:
       MOZ_ASSERT(false, "Can not map nsContentPolicyType to CSPDirective");
       // Do not add default: so that compilers can catch the missing case.
   }